flocker/ca/_ca.py

```diff
@@ -231,14 +231,81 @@ def write_credential_files(self, key_filename, certificate_filename):
             os.umask(original_umask)
 
 
-class NodeCredential(FlockerCredential):
+class UserCredential(PRecord):
+    """
+    A certificate for an API user, signed by a supplied certificate
+    authority.
+
+    :ivar FlockerCredential credential: The certificate and key pair
+        credential object.
+    :ivar bytes username: A username.
+    """
+    credential = field(mandatory=True)
+    username = field(mandatory=True, initial=None)
```

Comment:
You can do `type=bytes` or `type=FlockerCredential` in these field definitions to get type enforcement on them (and possibly make the unchecked type documentation in the class docstring superfluous).